smoke-tests/apps/SamplingOverrides/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/ExceptionSamplingOverridesTest.java

```diff
@@ -15,23 +15,36 @@
 import static org.assertj.core.api.Assertions.assertThat;
 
 import com.microsoft.applicationinsights.smoketest.schemav2.Envelope;
+import com.microsoft.applicationinsights.smoketest.schemav2.ExceptionData;
+import com.microsoft.applicationinsights.smoketest.schemav2.ExceptionDetails;
 import java.util.List;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
 @UseAgent("applicationinsights3.json")
 abstract class ExceptionSamplingOverridesTest {
 
-  @RegisterExtension static final SmokeTestExtension testing = SmokeTestExtension.create();
+  @RegisterExtension
+  static final SmokeTestExtension testing =
+      SmokeTestExtension.builder().setSelfDiagnosticsLevel("DEBUG").build();
 
+  @SuppressWarnings("SystemOut")
   @Test
   @TargetUri(value = "/trackException")
   void testExceptionSamplingOverrides() throws Exception {
     List<Envelope> rdList = testing.mockedIngestion.waitForItems("RequestData", 1);
     Envelope rdEnvelope = rdList.get(0);
     assertThat(rdEnvelope.getTags().get("ai.operation.name"))
         .isEqualTo("GET /SamplingOverrides/trackException");
-    assertThat(testing.mockedIngestion.getCountForType("ExceptionData")).isZero();
+    // Exception from the log will get sampled out and exception from the request won't.
+    List<ExceptionData> exceptions =
+        testing.mockedIngestion.getTelemetryDataByTypeInRequest("ExceptionData");
+    assertThat(exceptions.size()).isEqualTo(1);
+    assertThat(exceptions.get(0).getProperties().size()).isEqualTo(0);
+    ExceptionDetails exceptionDetails = exceptions.get(0).getExceptions().get(0);
+    assertThat(exceptionDetails.getStack()).isNotNull();
+    assertThat(exceptionDetails.getTypeName()).isEqualTo("java.lang.RuntimeException");
+    assertThat(exceptionDetails.getMessage()).isEqualTo("this is an expected exception");
```

Comment:
Exception will be zero when https://github.com/Azure/azure-sdk-for-java/pull/38687 is released.